[Bug]: `openai==1.47` breaks CI

### What happened?

Something about `openai==1.47.0` and `1.47.1` breaks [`paper-qa`](https://github.com/Future-House/paper-qa)'s CI. Our CI ([run](https://github.com/Future-House/paper-qa/actions/runs/10999639465)) starts blowing up with the below error.

### Relevant log output

```shell
litellm.exceptions.APIError: litellm.APIError: APIError: OpenAIException - Connection error.
```


### Twitter / LinkedIn details

_No response_

I was able to get a minimal repro for this issue:

```
openai==1.48.0
litellm==1.48.1
```
```python
import pytest
import litellm


@pytest.mark.asyncio
async def test_llm_completion1():
    await litellm.acompletion(
        model="gpt-4o-mini",
        temperature=0.5,
        messages=[
            {
                "content": "Here is a question, the correct answer to the question, and a proposed answer to the question. Please tell me if the proposed answer is correct, given the correct answer. ONLY SAY 'YES' OR 'NO'. No other output is permitted.\n\nQuestion: What is 25 * 10? \n\nCorrect answer: 250 \n\nProposed answer: 250",
                "role": "user",
            }
        ],
    )


@pytest.mark.asyncio
async def test_llm_completion2():
    await litellm.acompletion(
        model="gpt-4o-mini",
        temperature=0.5,
        messages=[
            {
                "content": "Here is a question, the correct answer to the question, and a proposed answer to the question. Please tell me if the proposed answer is correct, given the correct answer. ONLY SAY 'YES' OR 'NO'. No other output is permitted.\n\nQuestion: What is 25 * 10? \n\nCorrect answer: 250 \n\nProposed answer: 250",
                "role": "user",
            }
        ],
    )

```
and run with `pytest`

output (truncated)
```
FAILED tests/test_foo.py::test_llm_completion2 - litellm.exceptions.APIError: litellm.APIError: APIError: OpenAIException - Connection error.
```

>             log.debug("Raising timeout error")
            raise APITimeoutError(request=request) from err
        except Exception as err:
            log.debug("Encountered Exception", exc_info=True)
    
            if retries_taken > 0:
                return await self._retry_request(
                    input_options,
                    cast_to,
                    retries_taken=retries_taken,
                    stream=stream,
                    stream_cls=stream_cls,
                    response_headers=None,
                )
    
            log.debug("Raising connection error")
>           raise APIConnectionError(request=request) from err
E           openai.APIConnectionError: Connection error.

/usr/local/lib/python3.10/dist-packages/openai/_base_client.py:1598: APIConnectionError

this seems to come from openai

> self = <_UnixSelectorEventLoop running=False closed=True debug=False>

    def _check_closed(self):
        if self._closed:
>           raise RuntimeError('Event loop is closed')
E           RuntimeError: Event loop is closed

/usr/lib/python3.10/asyncio/base_events.py:515: RuntimeError


Is this somehow unique to being run in a notebook environment? 


It looks like something is closing the event loop prematurely

--- a/litellm/pytest_plugin.py
+++ b/litellm/pytest_plugin.py
@@ -1,0 +1,7 @@
+import pytest
+import litellm
+
+
+@pytest.fixture(autouse=True)
+def cleanup_caches() -> None:
+    litellm.in_memory_llm_clients_cache.flush_cache()

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -84,6 +84,9 @@ profile = "black"
 [tool.poetry.scripts]
 litellm = 'litellm:run_server'
 
+[tool.poetry.plugins.pytest11]
+pytest_litellm = "litellm.pytest_plugin"
+
 [tool.poetry.group.dev.dependencies]
 flake8 = "^6.1.0"
 black = "^23.12.0"

--- a/tests/pytest_plugin_tests/conftest.py
+++ b/tests/pytest_plugin_tests/conftest.py
@@ -1,0 +1,1 @@
+pytest_plugins = "pytester"

--- a/tests/pytest_plugin_tests/test_pytest_plugin.py
+++ b/tests/pytest_plugin_tests/test_pytest_plugin.py
@@ -1,0 +1,24 @@
+from pytest import Pytester
+
+
+class TestIsolationFixture:
+    CODE = """
+        import litellm
+        def test_write_cache_a():
+            assert litellm.in_memory_llm_clients_cache.get_cache("flushme") is None
+            litellm.in_memory_llm_clients_cache.set_cache(key="flushme", value="old_client")
+        
+        def test_write_cache_b():
+            assert litellm.in_memory_llm_clients_cache.get_cache("flushme") is None
+            litellm.in_memory_llm_clients_cache.set_cache(key="flushme", value="new_client")
+        """
+
+    def test_fixture_isolates(self, testdir: Pytester) -> None:
+        testdir.makepyfile(self.CODE)
+        result = testdir.runpytest("-p", "litellm.pytest_plugin")
+        result.assert_outcomes(passed=2)
+
+    def test_fixture_needed(self, testdir: Pytester) -> None:
+        testdir.makepyfile(self.CODE)
+        result = testdir.runpytest("-p", "no:litellm.pytest_plugin")
+        result.assert_outcomes(passed=1, failed=1)

--- a/tests/pytest_plugin_tests/test_pytest_plugin.py
+++ b/tests/pytest_plugin_tests/test_pytest_plugin.py
@@ -15,10 +15,15 @@ class TestIsolationFixture:
 
     def test_fixture_isolates(self, testdir: Pytester) -> None:
         testdir.makepyfile(self.CODE)
-        result = testdir.runpytest("-p", "litellm.pytest_plugin")
+        result = testdir.runpytest("-p", "pytest_litellm")
         result.assert_outcomes(passed=2)
 
     def test_fixture_needed(self, testdir: Pytester) -> None:
         testdir.makepyfile(self.CODE)
-        result = testdir.runpytest("-p", "no:litellm.pytest_plugin")
+        result = testdir.runpytest("-p", "no:pytest_litellm")
         result.assert_outcomes(passed=1, failed=1)
+
+    def test_fixture_used_automatically(self, testdir: Pytester) -> None:
+        testdir.makepyfile(self.CODE)
+        result = testdir.runpytest()
+        result.assert_outcomes(passed=2)

